hpcgap/lib/vecmat.gi

```diff
@@ -1730,6 +1730,71 @@ function(f,m)
   return Immutable(m);
 end);
 
+
+#############################################################################
+##
+#F  ImmutableVector( <field>, <vector>
+##
+InstallMethod( ImmutableVector,"general,2",[IsObject,IsRowVector],0,
+function(f,v)
+  local v2;
+  if not IsInt(f) then f := Size(f); fi;
+  if f <= 256 then
+    v2 := CopyToVectorRep(v,f);
+    if v2 <> fail then v := v2; fi;
+  fi;
+  return Immutable(v);
+end);
+
+InstallOtherMethod( ImmutableVector,"general,3",[IsObject,IsRowVector,IsBool],0,
+function(f,v,change)
+  local v2;
+  if not IsInt(f) then f := Size(f); fi;
+  if f <= 256 then
+    v2 := CopyToVectorRep(v,f);
+    if v2 <> fail then v := v2; fi;
+  fi;
+  return Immutable(v);
+end);
+
+InstallOtherMethod( ImmutableVector,"field,8bit",[IsField,Is8BitVectorRep],0,
+function(f,v)
+  if Q_VEC8BIT(v)<>Size(f) then
+    TryNextMethod();
+  fi;
+  return Immutable(v);
+end);
+
+InstallOtherMethod( ImmutableVector,"field,gf2",[IsField,IsGF2VectorRep],0,
+function(f,v)
+  if 2<>Size(f) then
+    TryNextMethod();
+  fi;
+  return Immutable(v);
+end);
```

Comment:
Is there any reason that with `RowVector` you do handle ints and fields with `if not IsInt(f) then f := Size(f); fi;`, but here you split into two separate methods?